Components/TeamList.js

```diff
@@ -0,0 +1,68 @@
+import useSWR from "swr";
+import styled from "styled-components";
+import Image from "next/image";
+
+const Header = styled.h1`
+  font-size: 5rem;
+  font-weight: bold;
+  margin-top: 1rem;  
+  font-family: "roboto", sans-serif;
+  text-align: center;
+  letter-spacing: 3px;
+ 
+`;
+
+const StyledList = styled.ul`
+  display: grid;
+  grid-template-columns: repeat(auto-fit, minmax(200px, 1fr));
+  grid-gap: 1.5rem;
+  list-style: none;
+  padding: 0;
+`;
+
+const StyledListItem = styled.li`
+  display: flex;
+  flex-direction: column;
+  justify-content: center;
+  align-items: center;
+  border-radius: 8px;
+  box-shadow: 0 4px 8px rgba(0, 0, 0, 0.1);
+  transition: all 0.2s ease-in-out;
+  cursor: pointer;
+
+  &:hover {
+    transform: translateY(-6px);
+    box-shadow: 0 8px 16px rgba(0, 0, 0, 0.2);
+  }
+`;
+
+const TeamName = styled.h2`
+  font-size: 1.3rem;
+  font-weight: bold;
+  margin-top: 1rem;
+  color: #0d48a0;
+  text-align: center;
+`;
+
+const URL = "https://www.balldontlie.io/api/v1/teams?page=1";
+export default function TeamOverview() {
+  const { data, error, isLoading } = useSWR(URL);
+  const teams = data?.data;
+
+ if (error) return <div>Failed to load</div>;
+ if (isLoading) return <div>Loading...</div>;
+
+  return (
+    <div>
+      <Header>TEAMS</Header>
+      <StyledList>
+      {teams.map((team) => (
+        <StyledListItem key={team.id}>
+          <Image src={`/images/team-logos/${team.id}.png`} width={200} height={200} style={{objectFit: "contain"}} alt={team.name}/>
+          <TeamName>{team.city} {team.name}</TeamName>
+        </StyledListItem>
+      ))} 
+      </StyledList>
+    </div>
```

Comment:
Just some little issues: 
- this structure screams out for a `<main>` element, instead of the generic `<div>` element 
- because you have set the list to `list-style: none`, some browsers won't recognize this as a list. This can lead to Accessibility problems. You can prevent this by adding a role to the list.

**Last but not least ☝️**
Since this is the TeamList component, I would render only the list here accordingly. 
➡️ Move the parent element (`<main>`) and the headline into the index.js.

```suggestion
    <main>
      <Title>TEAMS</Title>
      <StyledList role="list">
      {teams.map((team) => (
        <StyledListItem key={team.id}>
          <Image src={`/images/team-logos/${team.id}.png`} width={200} height={200} style={{objectFit: "contain"}} alt={team.name}/>
          <TeamName>{team.city} {team.name}</TeamName>
        </StyledListItem>
      ))} 
      </StyledList>
    </main>
```

Reply:
Thanks heaps for your detailed feedback! This surely helps a ton! =) 
 I've restructured the component accordingly.